src/m365/planner/commands/plan/plan-details-get.ts

```diff
@@ -0,0 +1,182 @@
+        args.options.planId = planId;
+        return this.getPlanDetails(args);
+      })
+      .then((res: PlannerPlanDetails): void => {
+        logger.log(res);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getGroupId(args: CommandArgs): Promise<string> {
+    if (args.options.planId) {
+      return Promise.resolve("");
+    }
+
+    if (args.options.ownerGroupId) {
+      return Promise.resolve(args.options.ownerGroupId);
+    }
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/groups?$filter=displayName eq '${encodeURIComponent(args.options.ownerGroupName as string)}'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: [{ id: string, resourceProvisioningOptions: string[] }] }>(requestOptions)
+      .then(response => {
+        const groupItem: { id: string, resourceProvisioningOptions: string[] } | undefined = response.value[0];
+
+        if (!groupItem) {
+          return Promise.reject(`The specified ownerGroup does not exist`);
+        }
+
+        if (response.value.length > 1) {
+          return Promise.reject(`Multiple ownerGroups with name ${args.options.ownerGroupName} found: Please choose between the following IDs ${response.value.map(x => x.id)}`);
+        }
+
+        return Promise.resolve(groupItem.id);
+      });
+  }
+
+  private getPlanId(args: CommandArgs): Promise<string> {
+    if (args.options.planId) {
+      return Promise.resolve(args.options.planId);
+    }
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/groups/${this.groupId}/planner/plans`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+    
+    return request
+      .get<{ value: PlannerPlan [] }>(requestOptions)
+      .then((response: { value: PlannerPlan[] }): Promise<string> => {
+        const filteredPlan = response.value.filter((plan: PlannerPlan) => plan.title === args.options.planTitle);
+        if (filteredPlan && filteredPlan.length > 0) {
+          if (filteredPlan.length > 1) {
+            return Promise.reject(`Multiple plans with name ${args.options.planTitle} found: ${filteredPlan.map(x => x.id)}`);
+          }
+          if(filteredPlan[0].id) {
+            return Promise.resolve(filteredPlan[0].id);
+          }
+        }
+        return Promise.reject(`The specified plan title does not exist`);
+      });    
+  }
+
+  private getPlanDetails(args: CommandArgs): Promise<PlannerPlanDetails> {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/plans/${args.options.planId}/details`,
+      headers: {
+        'accept': 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: PlannerPlanDetails }>(requestOptions)
```

Comment:
If you look at the [API docs](https://docs.microsoft.com/en-us/graph/api/plannerplandetails-get?view=graph-rest-1.0&tabs=http#response-1) you'll see that the response is of type `PlannerPlanDetails`, without being wrapped in `value`, so our code should be `return request.get<PlannerPlanDetails>(requestOptions)`

Reply:
Thanks for the correct code.